tests/ecmp/test_fgnhg.py

```diff
@@ -172,8 +176,31 @@ def partial_ptf_runner(ptfhost, test_case, dst_ip, exp_flow_count, **kwargs):
             qlen=1000,
             log_file=log_file)
 
+def reprogram_ecmp_routes(duthost, prefix_list, ip_to_port, ipcmd):
+    logger.info ("Reprogramming ECMP routes")
+    vtysh_base_cmd = "vtysh -c 'configure terminal'"
+    t_end = time.time() + 60 * 6
+    while time.time() < t_end:
+        for prefix in prefix_list:
+           cmd = vtysh_base_cmd
+           for nexthop in ip_to_port:
+               cmd = cmd + " -c '{} {} {}'".format(ipcmd, prefix, nexthop)
+           configure_dut(duthost, cmd)
+
+def hash_check_warm_boot(ptfhost, dst_ip_list, exp_flow_count):
+    logger.info ("I ran hash_check_warm_boot")
+    for dst_ip in dst_ip_list:
+        partial_ptf_runner(ptfhost, 'hash_check_warm_boot', dst_ip, exp_flow_count)    
+
+def continuous_hash_check_warm_boot(ptfhost, dst_ip_list, exp_flow_count):
+    t_end = time.time() + 60 * 6
+    while time.time() < t_end:
+        hash_check_warm_boot(ptfhost, dst_ip_list, exp_flow_count)         
+        logger.info ("hash check time is")
+        logger.info (time.asctime( time.localtime(time.time()) ))
```

Comment:
Same question as above. Why control plane down time didn't cause this thread to fail?

Reply:
The continuous vtysh performed by other concurrent thread should preserve data plane even during time period when the device is non-ssh'able. We have tested that it works on the mellanox DUT.